### PR TITLE
Fix Span not finished in gateway plugin when the gateway request time out

### PR DESCRIPTION
**Pull request by wuwen5**
- [x] fix https://github.com/apache/skywalking/issues/9061.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Release Notes.
 * Grpc plugin support trace client async generic call(without grpc stubs), support Method type: `UNARY`、`SERVER_STREAMING`.
 * Enhance Apache ShenYu (incubating) plugin: support trace `grpc`,`sofarpc`,`motan`,`tars` rpc proxy.
 * Add primary endpoint name to log events.
+* Fix Span not finished in gateway plugin when the gateway request timeout.
```
Comment (wu-sheng) [inline CHANGES.md:19]: Is this a Gateway 3.x plugin fix only? Or others may have similar issues?

File: test/plugin/scenarios/gateway-3.x-scenario/gateway-projectA-scenario/src/main/resources/application.yml [hunk 2]
```diff
@@ -20,8 +20,14 @@ server:
...
   cloud:
     gateway:
+      httpclient:
+        connect-timeout: 2000
       routes:
       - id: provider_route
         uri: http://localhost:18070
         predicates:
         - Path=/provider/b/*
+      - id: provider_timeout
```
Comment (wu-sheng) [inline test/plugin/scenarios/gateway-3.x-scenario/gateway-projectA-scenario/src/main/resources/application.yml:30]: You are adding a `timeout` calling, please update the expected data too.

File: CHANGES.md (hunk 1 above)
Comment (wuwen5) [reply CHANGES.md:19]: 2.1.x\3.x

**Review by wu-sheng — APPROVED** [review]
Thanks for fixing.

**Pull request merged**